### PR TITLE
Adds Metadata Artifacts to UI

### PR DESCRIPTION
**Pull request by rileyjbauer**
This change adds pages for listing and viewing metadata artifacts, as well as configuration for an envoy proxy that handles connections to the metadata gRPC server.

A follow-up will add pages for executions.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2057)

### CODE REVIEW

File: envoy.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+COPY envoy.yaml /etc/envoy.yaml
+COPY envoy-entrypoint.sh /
+
+RUN chmod 500 /envoy-entrypoint.sh
+
+RUN apt-get update && \
+    apt-get install gettext -y
+
+ENTRYPOINT ["/envoy-entrypoint.sh"]
```
Comment (neuromage) [inline envoy.Dockerfile]: I think we can dispense with the entrypoint script, and just run /usr/local/bin/envoy -c /etc/envoy.yaml here

Comment (rileyjbauer) [reply envoy.Dockerfile]: Sounds good. Will change this in the follow-up

**Pull request merged**